regression_tests/self_reference_or_path.jl

```diff
@@ -85,6 +85,10 @@ This is very similar to `self_reference_or_path`, except that it always returns
 function get_main_branch_buildkite_path()
     # TODO: Combine this method with `self_reference_or_path`, above.
     cluster_data_prefix = "/central/scratch/esm/slurm-buildkite/climaatmos-main"
+    if !ispath(cluster_data_prefix)
```

Comment:
```suggestion
    on_buildkite = haskey(ENV, "BUILDKITE_COMMIT") || haskey(ENV, "BUILDKITE_BRANCH")
    if !on_buildkite && !ispath(cluster_data_prefix)
```

Let's ensure that this errors somewhere rather than falling back on a local-run mode and silently passing if we're on buildkite, I think this should do it